graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+import warnings
+
+from graphql.type import GraphQLBoolean, GraphQLFloat, GraphQLID, GraphQLString
+import sqlalchemy.dialects.mssql.base as mssqltypes
+import sqlalchemy.sql.sqltypes as sqltypes
+
+from ...schema import GraphQLDate, GraphQLDateTime, GraphQLDecimal, GraphQLInt
+
+
+# The following quote from https://docs.sqlalchemy.org/en/13/core/type_basics.html
+# explains what makes the all-cap classes particular:
+# "This category of types refers to types that are either part of the SQL standard, or are
+# potentially found within a subset of database backends. Unlike the "generic" types, the SQL
+# standard/multi-vendor types have no guarantee of working on all backends, and will only work
+# on those backends that explicitly support them by name. That is, the type will always emit its
+# exact name in DDL with CREATE TABLE is issued."
+# TODO(pmantica1): Add scalar mapping for the following classes: Interval.
+GENERIC_SQL_CLASS_TO_GRAPHQL_TYPE = {
+    sqltypes.BIGINT: GraphQLInt,
+    sqltypes.BigInteger: GraphQLInt,
+    sqltypes.Boolean: GraphQLBoolean,
+    sqltypes.CHAR: GraphQLString,
+    sqltypes.CLOB: GraphQLString,
+    sqltypes.Date: GraphQLDate,
+    sqltypes.DATE: GraphQLDate,
+    sqltypes.DateTime: GraphQLDateTime,
+    sqltypes.DATETIME: GraphQLDateTime,
+    sqltypes.Enum: GraphQLString,
+    sqltypes.Float: GraphQLFloat,
+    sqltypes.FLOAT: GraphQLFloat,
+    sqltypes.INT: GraphQLInt,
+    sqltypes.Integer: GraphQLInt,
+    sqltypes.INTEGER: GraphQLInt,
+    sqltypes.NCHAR: GraphQLString,
+    sqltypes.Numeric: GraphQLDecimal,
+    sqltypes.NUMERIC: GraphQLDecimal,
+    sqltypes.NVARCHAR: GraphQLString,
+    sqltypes.REAL: GraphQLFloat,
+    sqltypes.SMALLINT: GraphQLInt,
+    sqltypes.SmallInteger: GraphQLInt,
+    sqltypes.String: GraphQLString,
+    sqltypes.Text: GraphQLString,
+    sqltypes.TEXT: GraphQLString,
+    sqltypes.Time: GraphQLDateTime,
+    sqltypes.TIME: GraphQLDateTime,
+    sqltypes.TIMESTAMP: GraphQLDateTime,
+    sqltypes.Unicode: GraphQLString,
+    sqltypes.UnicodeText: GraphQLString,
+    sqltypes.VARCHAR: GraphQLString,
+}
+
+# We do not currently plan to add a mapping for JSON and Binary objects.
+UNSUPPORTED_GENERIC_SQL_TYPES = frozenset({
```

Comment:
As @obi1kenobi told me in #313, [here](https://github.com/kensho-technologies/graphql-compiler/pull/313#commitcomment-33655951), creating an allowed (assuming GENERIC_SQL_CLASS_TO_GRAPHQL_TYPE means allowed) and disallowed list is sometimes redundant. In this case, can we create a set of allowed types, and disallow everything else?

Reply:
We are doing exactly that. We only have the unsupported types for documentation purposes. It's useful to keep track of the SQL classes for which we don't have a mapping.